src/helpers/utils.ts

```diff
@@ -82,3 +83,42 @@ export class StorageMap<T extends unknown> {
     }
   }
 }
+
+/**
+ * GraphQL returns the URL with protocol and the CMA without it,
+ * so we need to add the information in there manually.
+ */
+export function setProtocolToHttps(url: string | undefined): string | undefined {
+  if (!url) {
+    return url;
+  }
+
+  try {
+    // new URL('//images.ctfassets.net') is invalid, therefore we add it with protocol as `base`.
+    // This will merge it then correct together
+    const parsed = new URL(url, 'https://images.ctfassets.net');
+    parsed.protocol = 'https:';
+    return parsed.href;
+  } catch (err) {
+    debug.error('Recevied invalid asset url "${url}"', err);
+    return url;
+  }
+}
+
+/**
+ * Clones the incoming element into a new one, to prevent modification on the original object
+ * Hint: It uses the structuredClone which is only available in modern browsers,
+ * for older one it uses the JSON.parse(JSON.strinfify) hack.
+ */
+export function clone<T extends Record<string, unknown> | Array<unknown>>(incoming: T): T {
+  if (typeof structuredClone === 'function') {
+    return structuredClone(incoming);
+  }
+
+  try {
+    return JSON.parse(JSON.stringify(incoming));
```

Comment:
I think this will break if we have a Map in the object, you okay with that?

Reply:
I think it's fine for our update use case, just calling it out

Reply:
Yeah it's already the fallback, modern browsers (that support map) should also have this `structuredClone` function which is a native deep merge fn.
If that should not work, we try it with `JSON.parse(JSON.stringify)` thing, and otherwise we will work with the incoming one as laaaast fallback